src/plonk_challenger.rs

```diff
@@ -22,27 +29,42 @@ impl<F: Field> Challenger<F> {
         }
     }
 
-    pub(crate) fn observe_affine_point<C: Curve<BaseField = F>>(&mut self, point: AffinePoint<C>) {
-        debug_assert!(!point.zero);
-        self.observe_element(point.x);
-        self.observe_element(point.y);
+    pub(crate) fn observe_affine_point<C: Curve<BaseField=F>>(&mut self, point: AffinePoint<C>) {
```

Comment:
Your version seems like the natural thing to do, but it seems tricky to implement in the recursive version -- currently `AffinePointTarget` can't represent zero, and the curve-related gates can't handle zero.

Was there a case where zero was passed? If it was something like a Pedersen hash of the zero polynomial, I think we can work around that by adding a (constant) random point, maybe something hash-to-point(0) (we do that for `filler` in `curve_msm` on master).

Reply:
Yes I had a test that was failing because of this `assert`, but I can't remember which one, and now they all pass it. So I'll change it back like it was before.